apps/portals/src/portal-components/RouteControl.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useLayoutEffect, useRef } from 'react'
+import {
+  Tab,
+  Tabs,
+  TabScrollButton,
+  TabScrollButtonProps,
+  useMediaQuery,
+  useTheme,
+} from '@mui/material'
+
+export type RouteControlProps = {
+  handleChanges: (text: string, index: number) => void
+  isSelected: (name: string) => boolean
+  customRoutes: string[]
+}
+
+function CustomScrollButton(props: TabScrollButtonProps) {
+  if (props.disabled) {
+    return <></>
+  }
+  return (
+    <TabScrollButton
+      {...props}
+      sx={{
+        svg: {
+          color: 'secondary.main',
+          fontSize: '26px',
+        },
+        '&:hover': {
+          backgroundColor: 'grey.300',
+        },
+      }}
+    />
+  )
+}
+
+export function RouteControl(props: RouteControlProps) {
+  const { handleChanges, isSelected, customRoutes } = props
+  const theme = useTheme()
+  const isMobileView = useMediaQuery(theme.breakpoints.down('sm'))
+
+  const selectedRef = useRef<HTMLDivElement>(null)
+
+  useLayoutEffect(() => {
+    // setTimeout is necessary or else it only scrolls to reveal half of the button
+    setTimeout(() => {
+      selectedRef.current?.scrollIntoView(false)
+    }, 100)
+  }, [])
+
+  /**
+   * In the desktop view, we use Material UI tabs
+   */
+  return (
+    <Tabs
+      value={customRoutes.find((name) => isSelected(name))}
+      variant="scrollable"
+      orientation={isMobileView ? 'vertical' : 'horizontal'}
+      scrollButtons="auto"
+      ScrollButtonComponent={CustomScrollButton}
+      aria-label="Explore Sections"
+      sx={{
+        '.MuiTabs-flexContainer': {
+          gap: { xs: 2, sm: 5 },
```

Comment:
In `sx` we can use these breakpoints for responsive behavior. In this case:

If screen width < `sm`, gap of 2 spacing units, if width >= 'sm', it will be 5 spacing units.